roachtest: Update lib/pq block list

spilchen opened the pull request:
The recent autocommit_before_ddl change (#141145) has affected three lib/pq tests that were not designed with autocommit behavior in mind.

These tests were executing the following sequence:
```
BEGIN
CREATE TEMP TABLE
COPY
```

Since CREATE TEMP TABLE now triggers an auto-commit, the subsequent COPY command fails because it must run inside an active transaction. The error observed was: "pq: COPY is only allowed inside a transaction."

Epic: none
Release note: none

Closes #141899

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/142038)
<!-- Reviewable:end -->

annrpom reviewed (approved):
ty for taking these on; we also want to slap backport labels for this for when we merge the backports for the auto commit pr, right?

rafiss commented on pkg/cmd/roachtest/tests/libpq_blocklist.go:
```diff
@@ -30,6 +30,11 @@ var libPQBlocklist = blocklist{
 	"pq.TestStringWithNul":                           "26366",
+	// The following tests fail because they weren't designed for the
+	// autocommit_before_ddl behaviour in CRDB.
+	"pq.TestCopyInWrongType":      "unknown",
```
super nit: instead of "unknown", can we put in a short verbal description, or just this PR number so there's a breadcrumb

spilchen replied:
Makes sense. I couldn't find a suitable precedence for this sort of thing.
<!-- Reviewable comment -OK2UExSElXVjvRrfbwY:b-8sdu9u -->
<!-- Sent from Reviewable.io -->

spilchen committed:
roachtest: Update lib/pq block list

The recent autocommit_before_ddl change (#141145) has affected three
lib/pq tests that were not designed with autocommit behavior in mind.

These tests were executing the following sequence:
```
BEGIN
CREATE TEMP TABLE
COPY
```

Since CREATE TEMP TABLE now triggers an auto-commit, the subsequent COPY
command fails because it must run inside an active transaction. The
error observed was: "pq: COPY is only allowed inside a transaction."

Epic: none
Release note: none

Closes #141899

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142038)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss)



<!-- Sent from Reviewable.io -->

spilchen commented:
> ty for taking these on; we also want to slap backport labels for this for when we merge the backports for the auto commit pr, right?

Yes, good point. I'll get this back ported. I'll leave the labels off this PR because I don't want blathers to create a new PR. Instead I'll cherry pick the comment into these two PRs: #141851 and #141987

spilchen commented:
TFTR!

bors r+
